tests/scripts/thread-cert/Cert_5_3_10_AddressQuery.py

```diff
@@ -91,52 +105,125 @@ def test(self):
+
+        # Verify the DUT_ROUTER2 forwarded ICMPv6 Echo Request to ROUTER1
+        msg = dut_router2_messages.get_icmp_message(ipv6.ICMP_ECHO_REQUEST)
+        assert msg is not None, "Error: The DUT_ROUTER2 didn't forward ICMPv6 Echo Request to ROUTER1"
+        msg.assertSentToNode(self.nodes[ROUTER1])
+
+        # 4 BR: BR sends an ICMPv6 Echo Request to MED1 using GUA 2003:: address
+        addr = self.nodes[MED1].get_addr("2003::/64")
+        self.assertTrue(addr is not None)
+        self.assertTrue(self.nodes[BR].ping(addr))
+
+        # Wait for sniffer got Address Notification messages
+        time.sleep(1)
+
+        # Verify DUT_ROUTER2 sent an Address Notification message
+        dut_router2_messages = self.sniffer.get_messages_sent_by(DUT_ROUTER2)
+        msg = dut_router2_messages.next_coap_message('0.02', '/a/an')
+
+        # Verify the IPv6 Source address is the Rloc of the DUT_ROUTER2
+        dut_router2_rloc = ipaddress.ip_address(self.nodes[DUT_ROUTER2].get_ip6_address(config.ADDRESS_TYPE.RLOC))
+        assert(dut_router2_rloc == msg.ipv6_packet.ipv6_header.source_address)
+
+        # Verify the IPv6 Destination address is the Rloc of the BR
+        br_rloc = ipaddress.ip_address(self.nodes[BR].get_ip6_address(config.ADDRESS_TYPE.RLOC))
+        assert(br_rloc == msg.ipv6_packet.ipv6_header.destination_address)
+
+        # 5 MED1: MED1 sends an ICMPv6 Echo Request to ROUTER1 using GUA 2003:: address
+        addr = self.nodes[ROUTER1].get_addr("2003::/64")
+        self.assertTrue(addr is not None)
+        self.assertTrue(self.nodes[MED1].ping(addr))
+
+        # Wait for sniffer got ICMPv6 Echo Reply
+        time.sleep(1)
+
+        # Verify DUT_ROUTER2 didn't generate an Address Query Request
+        dut_router2_messages = self.sniffer.get_messages_sent_by(DUT_ROUTER2)
+        dut_router2_messages_temp = copy.deepcopy(dut_router2_messages)
+        msg = dut_router2_messages.next_coap_message('0.02', '/a/aq', False)
+        assert msg is None, "Error: The DUT_ROUTER2 sent an unexpected Address Query Request"
+
+        # Verify DUT_ROUTER2 forwarded ICMPv6 Echo Reply to MED1
+        msg = dut_router2_messages_temp.get_icmp_message(ipv6.ICMP_ECHO_RESPONSE)
+        assert msg is not None, "Error: The DUT_ROUTER2 didn's forward ICMPv6 Echo Reply to MED1"
+        msg.assertSentToNode(self.nodes[MED1])
+
+        # 6 DUT_ROUTER2: Power off ROUTER1 and wait 580 seconds to allow the LEADER to expire its Router ID
+        self.nodes[MED1].get_addrs()
+        router1_addr = self.nodes[ROUTER1].get_addr("2003::/64")
+        router1_id = self.nodes[ROUTER1].get_addr16()
+
+        self.nodes[ROUTER1].stop()
+        time.sleep(580)
+
+        # Send an ICMPv6 Echo Request from SED1 to ROUTER1 GUA 2003:: address
+        self.assertTrue(router1_addr is not None)
+        self.assertFalse(self.nodes[MED1].ping(router1_addr))
+
+        # Verify the ROUTER2 has removed all entries based on ROUTER1's Router ID
```

Comment:
`router1_id` you are using is actually rloc16 address, not `ROUTER1's Router ID`
nit: ROUTER2 -> DUT_ROUTER2

Reply:
Will update it